nessus_ec2.tf

```diff
@@ -0,0 +1,53 @@
+# The Nessus AMI
+data "aws_ami" "nessus" {
+  provider = aws.provisionassessment
+
+  filter {
+    name = "name"
+    values = [
+      "nessus-hvm-*-x86_64-ebs"
+    ]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "root-device-type"
+    values = ["ebs"]
+  }
+
+  owners      = [local.images_account_id]
+  most_recent = true
+}
+
+# The Nessus EC2 instance
+resource "aws_instance" "nessus" {
+  count    = var.operations_instance_counts["nessus"]
+  provider = aws.provisionassessment
+
+  ami                         = data.aws_ami.nessus.id
+  associate_public_ip_address = true
+  availability_zone           = "${var.aws_region}${var.aws_availability_zone}"
+  iam_instance_profile        = aws_iam_instance_profile.nessus.name
+  instance_type               = "m5.large"
+  subnet_id                   = aws_subnet.operations.id
+
+  root_block_device {
+    volume_type           = "gp2"
+    volume_size           = 128
+    delete_on_termination = true
+  }
+
+  user_data_base64 = data.template_cloudinit_config.nessus_cloud_init_tasks[count.index].rendered
+
+  vpc_security_group_ids = [
+    aws_security_group.efs_client.id,
```

Comment:
You put the Nessus instances in this security group, so I assume you intend for it to be able to mount the EFS file share.  If that is true, then you need to add a policy to allow that to the instance role.  See [here](https://github.com/cisagov/cool-assessment-terraform/blob/develop/kali_iam.tf#L36-L43) and [here](https://github.com/cisagov/cool-assessment-terraform/blob/develop/kali_iam.tf#L60-L81) for an example.  You will also need to add some cloud-init foo that is similar to [this](https://github.com/cisagov/cool-assessment-terraform/blob/develop/kali_cloud_init.tf#L13-L23) to make sure the mount point is created and the EFS file system is added to `/etc/fstab`.

If you _do not_ intend the Nessus instances to mount the EFS file share then you should remove them from this security group.

Reply:
@jsf9k - I did not intend to give the Nessus instances access to the EFS share.  That was a copy/paste mistake on my part.

My reasoning was that if Nessus is only being accessed via the web GUI (from the Kali instances), there is no need for the Nessus instance to ever use the EFS share.  Can you think of any good reasons why it might be needed?

Reply:
No, I just wasn't sure which way you were going with it.

Reply:
Thanks for catching this- I removed the Nessus instances from the EFS security group in https://github.com/cisagov/cool-assessment-terraform/pull/31/commits/b41b84f766d3b0ab892cfc252743434302dab332.